src/MapfileStyleParser.ts

```diff
@@ -979,11 +982,7 @@ export class MapfileStyleParser implements StyleParser {
       case ')':
         parantheseCount--;
         if (parantheseCount === 0) {
-          const operators: string = nestedExpressions[0]
-            ? mapfileExpression.substring(nestedExpressions[0].length, fromIndex)
-            : mapfileExpression.substring(0, fromIndex);
-          const negationOperator: string = operators.replace(/\s*(AND|&&|OR|\|\|)\s*/, '');
-          nestedExpressions.push(negationOperator + mapfileExpression.substring(fromIndex, index + 1));
+          nestedExpressions.push(mapfileExpression.substring(fromIndex, index + 1));
```

Comment:
Is the operator not needed anymore?

Reply:
The `negationOperator` led to an incorrect interpretation or composition of the defined filters.
For example 

`EXPRESSION (('[class]' = 'minorroads') AND ('[tracktype]' != 'grade1') AND ('[type]' = 'track'))`

resulted in

```javscript
"&&",
["==", "class", "minorroads"],
["!=", "tracktype", "grade1"],
["!=", "tracktype", "grade1"]
```

Removing the `negationOperator` resulted in the correct interpretation:

```javscript
"&&",
["==", "class", "minorroads"],
["!=", "tracktype", "grade1"],
["==", "type", "track"]
```

Reply:
Thx for the explanation.